src/Main.purs

```diff
@@ -260,7 +261,18 @@ runRun cliRoot args = do
 evalScheme :: Array String -> String -> Aff ExecaResult
 evalScheme arguments code = do
   schemeBin <- getSchemeBinary
-  spawned <- execa schemeBin arguments identity
+
+  -- Apple blocks DYLD environment variables, so loading libraries with DYLD_LIBRARY_PATH
+  -- will not work across process spawns. We go around this by carrying the library path
+  -- in CHEZ_DYLD_LIBRARY_PATH and setting DYLD_LIBRARY_PATH when spawning. For more info see:
+  -- https://developer.apple.com/library/archive/documentation/Security/Conceptual/System_Integrity_Protection_Guide/RuntimeProtections/RuntimeProtections.html
+  env <- liftEffect $ Process.getEnv
+  case Object.lookup "CHEZ_DYLD_LIBRARY_PATH" env of
+    Just dyldPath -> do
+      liftEffect $ Process.setEnv "DYLD_LIBRARY_PATH" dyldPath
```

Comment:
Why not just read and set `DYLD_LIBRARY_PATH` directly instead of reading from `CHEZ_DYLD_LIBRARY_PATH`. The only case I can think of is if you want to link to different versions of some library in the parent process and `scheme`.

Reply:
We can't just read `DYLD_LIBRARY_PATH` because it will be stripped away when launching `purescm` at the command line - from the link above:
> environment variables, such as DYLD_LIBRARY_PATH, are purged when launching protected processes